okhttp-tests/src/test/java/okhttp3/RequestTest.java

```diff
@@ -242,6 +243,103 @@ private void assertForbiddenHeader(String s) {
     }
   }
 
+  @Test public void noTag() {
+    Request request = new Request.Builder()
+        .url("https://square.com")
+        .build();
+    assertNull(request.tag());
+    assertNull(request.tag(Object.class));
+    assertNull(request.tag(UUID.class));
+    assertNull(request.tag(String.class));
+  }
+
+  @Test public void defaultTag() {
+    UUID tag = UUID.randomUUID();
+    Request request = new Request.Builder()
+        .url("https://square.com")
+        .tag(tag)
+        .build();
+    assertEquals(tag, request.tag());
+    assertEquals(tag, request.tag(Object.class));
```

Comment:
If I'm being super nit picky these should all be assertSame

Reply:
Good call. Fixed all.